code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -189,11 +214,19 @@
 	var/datum/robot_component/C = components[module_name]
 	return C && C.installed && C.toggled && C.is_powered() && !C.component_disabled
 
+///Disables a random component for the duration, or until manually turned back on.
+/mob/living/silicon/robot/proc/disable_random_component(number_disabled, duration)
+	var/list/random_components = pick_multiple_unique(components, number_disabled)
+	for(var/component in random_components)
+		disable_component(component, duration)
```

Comment:
what happens if it picks the cell?

Reply:
I overwrote the cell's Disable() proc, so it does nothing

Reply:
so this proc has a chance of doing nothing?